packages/kit/src/core/dev/plugin.js

```diff
@@ -203,7 +204,13 @@ export async function create_plugin(config, cwd) {
 
 						if (req.url === '/favicon.ico') return not_found(res);
 
-						if (!decoded.startsWith(config.kit.paths.base)) return not_found(res);
+						if (!decoded.startsWith(config.kit.paths.base)) {
+							const suggestion = normalize_path(
+								config.kit.paths.base + req.url,
+								config.kit.trailingSlash
+							);
+							return not_found(res, `Not found (did you mean ${suggestion}?)`);
```

Comment:
Perhaps?
```suggestion
							return not_found(res, `Not found (did you mean <a href="${suggestion}">${suggestion}</a>?)`);
```

Reply:
This error message gets returned in a plain text response (with no `content-type`). I don't think it's worth it to make it an HTML response and then worry about escaping the suggestion appropriately.

Reply:
yeah, that's the journey i went on too